internal/manifests/collector/service_test.go

```diff
@@ -321,6 +322,206 @@ func TestMonitoringService(t *testing.T) {
 	})
 }
 
+func TestExtensionService(t *testing.T) {
+	testCases := []struct {
+		name          string
+		params        manifests.Params
+		expectedPorts []v1.ServicePort
+	}{
+		{
+			name: "when the extension has http endpoint",
+			params: manifests.Params{
+				Config: config.Config{},
+				Log:    logger,
+				OtelCol: v1beta1.OpenTelemetryCollector{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test",
+					},
+					Spec: v1beta1.OpenTelemetryCollectorSpec{
+						Config: v1beta1.Config{
+							Service: v1beta1.Service{
+								Extensions: []string{"jaeger_query"},
+							},
+							Extensions: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"jaeger_query": map[string]interface{}{
+										"http": map[string]interface{}{
+											"endpoint": "0.0.0.0:16686",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedPorts: []v1.ServicePort{
+				{
+					Name: "jaeger-query",
+					Port: 16686,
+					TargetPort: intstr.IntOrString{
+						IntVal: 16686,
+					},
+				},
+			},
+		},
+		{
+			name: "when the extension has grpc endpoint",
+			params: manifests.Params{
+				Config: config.Config{},
+				Log:    logger,
+				OtelCol: v1beta1.OpenTelemetryCollector{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test",
+					},
+					Spec: v1beta1.OpenTelemetryCollectorSpec{
+						Config: v1beta1.Config{
+							Service: v1beta1.Service{
+								Extensions: []string{"jaeger_query"},
+							},
+							Extensions: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"jaeger_query": map[string]interface{}{
+										"http": map[string]interface{}{
+											"endpoint": "0.0.0.0:16686",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedPorts: []v1.ServicePort{
+				{
+					Name: "jaeger-query",
+					Port: 16686,
+					TargetPort: intstr.IntOrString{
+						IntVal: 16686,
+					},
+				},
+			},
+		},
+		{
+			name: "when the extension has both http and grpc endpoint",
+			params: manifests.Params{
+				Config: config.Config{},
+				Log:    logger,
+				OtelCol: v1beta1.OpenTelemetryCollector{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test",
+					},
+					Spec: v1beta1.OpenTelemetryCollectorSpec{
+						Config: v1beta1.Config{
+							Service: v1beta1.Service{
+								Extensions: []string{"jaeger_query"},
+							},
+							Extensions: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"jaeger_query": map[string]interface{}{
+										"http": map[string]interface{}{
+											"endpoint": "0.0.0.0:16686",
+										},
+										"grpc": map[string]interface{}{
+											"endpoint": "0.0.0.0:16686",
```

Comment:
wouldnt we expect grpc and http to not share an endpoint?

Reply:
Actually, `jaeger-query` won't be using GRPC endpoint. I have intentionally kept that to see if it is throwing an error or not.

Reply:
got it, @yurishkuro would you mind reviewing this and then i can merge it in?

Reply:
@jaronoff97 you should not block on my review - I know nothing about how operators are implemented internally, and this change is about that, not about any Jaeger knowledge, so my review would be completely superficial.

Reply:
i mostly wanted the jaeger knowledge and the superficial ✔️ just to close out the comments you left from before. but no problem, thanks for letting me know :)